react/components/EditorContainer/Topbar/index.tsx

```diff
@@ -9,43 +12,94 @@ interface Props {
   changeMode: (mode?: StoreEditMode) => void
   mode?: StoreEditMode
   urlPath: string
+  onChangeUrlPath: (url: string) => void
   visible: boolean
 }
 
 const Topbar: React.FunctionComponent<Props> = ({
   changeMode,
   mode,
+  onChangeUrlPath,
   urlPath,
   visible,
-}) => (
-  <div
-    className={
-      visible ? 'ph5 f6 h-3em w-100 flex justify-between items-center' : 'dn'
+}) => {
+  const editor = useEditorContext()
+  const [urlInputDisabled, setUrlInputDisabled] = useState(false)
+  const [url, setUrl] = useState(urlPath)
+
+  useEffect(
+    () => {
+      setUrl(urlPath)
+    },
+    [urlPath]
+  )
+
+  const onEnter = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+    callback: (...args: any[]) => any
```

Comment:
Please avoid using `any`.
Since you only use `onEnter` with `handleNavigateToUrl`, you could skip passing the parameter and use a closure.
But if you want to keep it like this, `handleNavigateToUrl` has type `() => void`.